### PR TITLE
LABS-1268 Fix crash on non block meta and oligarch delete bugs

### PR DESCRIPTION
**Pull request by quchenhao**
Summarize the reason behind this change (this might be the problem you're solving, or the context around the request) and the solution you have chosen.

### CODE REVIEW

File: src/oligarch_log/oligarch_log.c [hunk 1]
```diff
@@ -877,13 +877,12 @@ __oligarch_log_openfile(WT_SESSION_IMPL *session, uint32_t id, uint32_t flags, W
         FLD_SET(wtopen_flags, WT_FS_OPEN_DIRECTIO);
 
     /*
-     * XXX
-     * Open failures of log files may occur in disaggregated storage.  Suppress the errors for now.
-     * Eventually we expect the oligarch log to not use files anyway.
-     * Once we do that, remove the new session flag to quiet open file failures.
+     * XXX Open failures of log files may occur in disaggregated storage. Suppress the errors for
```
Comment (quchenhao) [inline src/oligarch_log/oligarch_log.c:880]: This change is unrelated. s_all makes this change.

File: src/reconcile/rec_write.c [hunk 2]
```diff
@@ -2830,14 +2831,22 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
...
          * If this is the root page, we need to create a sync point. For a page to be empty, it has
          * to contain nothing at all, which means it has no records of any kind and is durable.
          */
         ref = r->ref;
         if (__wt_ref_is_root(ref)) {
             __wt_checkpoint_tree_reconcile_update(session, &ta);
-            WT_RET(bm->checkpoint(bm, session, NULL, NULL, btree->ckpt, false));
+            WT_RET(bm->checkpoint(
+              bm, session, NULL, &r->wrapup_checkpoint_block_meta, btree->ckpt, false));
+            r->ref->page->block_meta = r->wrapup_checkpoint_block_meta;
```
Comment (ddanderson) [inline src/reconcile/rec_write.c:2849]: How does the original `page->block_meta` get freed?

**Review by ddanderson — APPROVED** [review]
LGTM - just a question about the lifetime of the block meta page.

File: src/reconcile/rec_write.c (hunk 2 above)
Comment (quchenhao) [reply src/reconcile/rec_write.c:2849]: It is part of the page and it is freed when page is evicted.
